ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -12550,6 +12550,20 @@ private ParseResult rewriteASTWithMaskAndFilter(TableMask tableMask, ASTNode ast
     }
   }
 
+  void gatherUserSuppliedFunctions(ASTNode ast) {
+    int tokenType = ast.getToken().getType();
+    if (tokenType == HiveParser.TOK_FUNCTION ||
+            tokenType == HiveParser.TOK_FUNCTIONDI ||
+            tokenType == HiveParser.TOK_FUNCTIONSTAR) {
+      if (ast.getChild(0).getType() == HiveParser.Identifier) {
+        this.userSuppliedFunctions.add(unescapeIdentifier(ast.getChild(0).getText()));
```

Comment:
Could we add the lower-cased function names into `userSuppliedFunctions`? I wonder there are some queries like: `select MIN(a) from table_example`.
Does it handle cast properly? for example: `select cast(a as int) from `table_example`.

Reply:
Ack